### PR TITLE
WIP: use babel v7

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,9 @@
   "peerDependencies": {
     "eslint": ">=3.1.0"
   },
-  "dependencies": {},
+  "dependencies": {
```
Comment (graingert) [inline package.json]: woops this should be devDeps.

File: tests/lib/rules/no-useless-rest-spread.js [hunk 2]
```diff
@@ -127,11 +115,27 @@ ruleTester.run("no-useless-rest-spread", rule, {
...
+            output: "let {a, b, c, ...d} = obj",
+            errors: ["Unexpected a rest property."],
+        },
+        {
+            code: "let {a, ...{b, c, }} = obj",
+            output: "let {a, b, c} = obj",
+            errors: ["Unexpected a rest property."],
+        },
+        {
+            code: "let {a, ...{}} = obj",
```
Comment (mysticatea) [inline tests/lib/rules/no-useless-rest-spread.js]: This is a syntax error in the final spec. `ObjectPattern` and `ArrayPattern` cannot be after rest operator.

- https://github.com/tc39/proposal-object-rest-spread/pull/50
- https://github.com/tc39/proposal-object-rest-spread/pull/51

Comment (graingert) [reply tests/lib/rules/no-useless-rest-spread.js]: Probably worth just ditching the tests then?

Comment (mysticatea) [reply tests/lib/rules/no-useless-rest-spread.js]: ESLint's test runner cannot include test cases which include syntax errors. Those are failed always even if it's in invalid tests. So I merely removed those commented-out codes in the master branch.

Comment (graingert) [reply tests/lib/rules/no-useless-rest-spread.js]: @mysticatea cool, I closed the PR

**Pull request closed without merging**